mesheryctl/internal/cli/root/system/start.go

```diff
@@ -86,167 +89,234 @@ func start() error {
-		return errors.Wrap(err, utils.SystemError("failed to create new env client"))
-	}
+		//fmt.Println("Services", services);
+		//fmt.Println("RequiredAdapters", RequestedAdapters);
+		//fmt.Println("AllowedServices", AllowedServices);
+		//fmt.Println("version", utils.ViperCompose.GetString("version")) // Works here
 
-	containers, err := cli.ContainerList(context.Background(), types.ContainerListOptions{})
-	if err != nil {
-		return errors.Wrap(err, utils.SystemError("failed to fetch the list of containers"))
-	}
+		//////// FLAGS
+		// Control whether to pull for new Meshery container images
+		if skipUpdateFlag {
+			log.Info("Skipping Meshery update...")
+		} else {
+			err := utils.UpdateMesheryContainers()
+			if err != nil {
+				return errors.Wrap(err, utils.SystemError("failed to update meshery containers"))
+			}
+		}
 
-	//check for container meshery_meshery_1 running status
-	for _, container := range containers {
-		if container.Names[0] == "/meshery_meshery_1" {
-			log.Info("Opening Meshery in your browser. If Meshery does not open, please point your browser to http://localhost:9081 to access Meshery.")
-
-			//check for os of host machine
-			if runtime.GOOS == "windows" {
-				// Meshery running on Windows host
-				err = exec.Command("rundll32", "url.dll,FileProtocolHandler", url).Start()
-				if err != nil {
-					return errors.Wrap(err, utils.SystemError("failed to exec command"))
-				}
-			} else if runtime.GOOS == "linux" {
-				// Meshery running on Linux host
-				_, err = exec.LookPath("xdg-open")
-				if err != nil {
-					break
-				}
-				err = exec.Command("xdg-open", url).Start()
-				if err != nil {
-					return errors.Wrap(err, utils.SystemError("failed to exec command"))
+		// Reset Meshery config file to default settings
+		if utils.ResetFlag {
+			err := resetMesheryConfig()
+			if err != nil {
+				return errors.Wrap(err, utils.SystemError("failed to reset meshery config"))
+			}
+		}
+
+		log.Info("Starting Meshery...")
+		start := exec.Command("docker-compose", "-f", utils.DockerComposeFile, "up", "-d")
+		start.Stdout = os.Stdout
+		start.Stderr = os.Stderr
+
+		if err := start.Run(); err != nil {
+			return errors.Wrap(err, utils.SystemError("failed to run meshery server"))
+		}
+		checkFlag := 0 //flag to check
+
+		//connection to docker-client
+		cli, err := client.NewClientWithOpts(client.FromEnv)
+		if err != nil {
+			return errors.Wrap(err, utils.SystemError("failed to create new env client"))
+		}
+
+		containers, err := cli.ContainerList(context.Background(), types.ContainerListOptions{})
+		if err != nil {
+			return errors.Wrap(err, utils.SystemError("failed to fetch the list of containers"))
+		}
+
+		//check for container meshery_meshery_1 running status
+		for _, container := range containers {
+			if container.Names[0] == "/meshery_meshery_1" {
+				log.Info("Opening Meshery in your browser. If Meshery does not open, please point your browser to http://localhost:9081 to access Meshery.")
```

Comment:
We'll want to use variables to refer to Meshery's URL and port. These will be somewhat environment specific to their Kubernetes deployment.

Reply:
Yes. I have not worked on the docker-compose side of things, just moved everything into a switch-case. But we should definitely add this. I will add it along with the context support. I will keep this in mind.